catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/dialog/login/RegistrationDialogFragment.java

```diff
@@ -125,16 +125,22 @@ private void onRegisterButtonClick() {
 		String passwordConfirmation = confirmPasswordInputLayout.getEditText().getText().toString();
 		String email = emailInputLayout.getEditText().getText().toString();
 
-		if (password.equals(passwordConfirmation)) {
+		if (username.contains("@")) {
+			showErrorAlertDialog(R.string.register_email_as_username_error);
+		} else if (password.equals(passwordConfirmation)) {
 			RegistrationTask registrationTask = new RegistrationTask(getActivity(), username, password, email);
 			registrationTask.setOnRegistrationCompleteListener(this);
 			registrationTask.execute();
 		} else {
-			new AlertDialog.Builder(getActivity())
-					.setTitle(R.string.register_error)
-					.setMessage(R.string.register_password_mismatch)
-					.setPositiveButton(R.string.ok, null)
-					.show();
+			showErrorAlertDialog(R.string.register_password_mismatch);
```

Comment:
Please change the order of the if-else statements
```
if ("@" in username)
  show error
else if (not password equals) 
  show error
else
  execute registration task
```

Reply:
done.